silkworm/db/datastore/snapshots/rec_split/rec_split.hpp

```diff
@@ -706,10 +706,10 @@ class RecSplit {
         return lookup_by_ordinal(Ordinal{data_id - base_data_id()});
     }
 
-    std::optional<size_t> lookup_by_key(ByteView key) const {
+    std::optional<std::pair<size_t, uint64_t>> lookup_by_key(ByteView key) const {
```

Comment:
instead of modifying this method I suggest to call 2 other methods in a sequence from FindByHashSegmentQuery:
1. lookup_data_id_by_key - returns a data_id (Timestamp) by key
2. lookup_by_data_id - returns a value (offset in .seg) by data_id (Timestamp)

lookup_data_id_by_key needs to be exposed on the AccessorIndex (see accessor_index.hpp using declarations)

Reply:
the methods I've mentioned (lookup_data_id_by_key/lookup_by_data_id) already do the job, and one is already exposed, no need to expose more or replicate anything, it will just work, trust me :)